### PR TITLE
feat(searchbar): add name property

### LINKED ISSUE #27675: feat: ion-searchbar should have a 'name' property
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

The `ion-searchbar` component should have a `name` property spread to its input hidden field.

### Describe the Use Case

In this way when the searchbar is used with a form, its value is posted too.

In my scenario I have a searchbar in a form. The searchbar loads data via ajax. When the user selects one of the proposed items, the selected one must be posted when the form is submitted.

### Describe Preferred Solution

_No response_

### Describe Alternatives

_No response_

### Related Code

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by NicoDos**
Issue number: Resolves #27675

---------

## What is the current behavior?
There is no name property to ion-search

## What is the new behavior?
- Added name property to ion-search

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

### CODE REVIEW

**Comment by liamdebeasi** [comment]
Thanks for the PR! We appreciate you contributing to Ionic. In terms of next steps, the team needs to do some design work to make sure this `name` property aligns with the `name` properties in other components. Once that is complete, someone from the team will do a code review and request any changes. Let me know if you have any questions.

File: core/src/components/searchbar/searchbar.tsx [hunk 1]
```diff
@@ -111,6 +111,11 @@ export class Searchbar implements ComponentInterface {
    */
   @Prop() enterkeyhint?: 'enter' | 'done' | 'go' | 'next' | 'previous' | 'search' | 'send';
 
+  /**
+   * If used in a form, set the name of the control, which is submitted with the form data.
+   */
+  @Prop() name?: string;
```
Comment (sean-perkins) [inline core/src/components/searchbar/searchbar.tsx:117]: We'd like to a follow a similar architecture that we've done here with `ion-radio`: https://github.com/ionic-team/ionic-framework/blob/3bd1d7e1e501ca82830de1b375b38135ab4f6311/core/src/components/radio/radio.tsx#L62

Basically, defaulting the value of `name` to the `inputId`. 

The `inputId` would be a new member of this class:

```ts
private inputId = `ion-searchbar-${searchbarIds++}`;
```

where `let searchbarIds = 0;` is added outside of the search bar class, at the top or bottom of this file.

**Review by sean-perkins — CHANGES_REQUESTED** [review]
I only have one outstanding change + re-generating the proxies after syncing with `main`.

**Comment by sean-perkins** [comment]
Ok - this PR is ready for a team review. `feature-7.2.` is fairly behind `main`, so I'll wait to changing the target branch until we get that branch synced.

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**